utils.go

```diff
@@ -1,27 +1,108 @@
 package ipfs
 
 import (
+	"errors"
+	"fmt"
+
+	ipfs "github.com/ipfs/go-ipfs-api"
+
+	"github.com/beyondstorage/go-endpoint"
+	"github.com/beyondstorage/go-storage/v4/services"
 	"github.com/beyondstorage/go-storage/v4/types"
 )
 
 // Storage is the example client.
 type Storage struct {
+	ipfs *ipfs.Shell
+
 	defaultPairs DefaultStoragePairs
 	features     StorageFeatures
 
+	name    string
```

Comment:
Can you explain the meaning of `name` to me? It seems not to be used practically. 

e.g., in object storage, `name` is the bucket's name, and in fs, there's no `name`. What is it like in IPFS here?